### PR TITLE
Product details: Decouple image gallery from product details

### PR DESCRIPTION
**Pull request by 0nko**
This PR decouples the image gallery from the `ProductDetailsViewModel` so that it can be reused in Variation details. The image gallery functionality should not change.

**To test:**
1. Go to Products -> Tap on a product item
2. Smoke test:
  - adding new images
  - removing images
  - discarding current uploads
3. Verify the update button is displayed correctly when there are changes

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductImagesFragment.kt [hunk 1]
```diff
@@ -6,167 +6,159 @@ import android.content.Intent
...
-    }
-
     override fun onPause() {
         super.onPause()
         imageSourceDialog?.dismiss()
     }
 
-    override fun onResume() {
-        super.onResume()
-        AnalyticsTracker.trackViewShown(this)
```
Comment (AmandaRiu) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductImagesFragment.kt:90]: Looks like this track event is now missing. Was this intentional?

Comment (AmandaRiu) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductImagesFragment.kt:90]: Looks like this track event is now missing. Was this intentional?

**Review by AmandaRiu — CHANGES_REQUESTED** [review]
@0nko Nice work on this. I've done a bunch of tests and ran into the following:

Disappearing upload thumbnail. If uploading an image to a product has been canceled, adding images after will not be displayed in the list of product photos while uploading and will only suddenly appear once the upload is complete. Here are the steps:

1. Open a product and click to add an image to the product
2. Select an image from the device…note the *loading* thumbnail displayed showing the photo is being uploaded.
3. Click the **back** button and **discard** the changes to the product.
4. Click back into the images to add another image from the device.
5. Select an image from the device…note the *loading* thumbnail is **not** displayed and the photo only appears in the image list **after** the upload is complete.

ezgif com-video-to-gif (1)


Also, after completing the steps above I’m seeing LeakCanary leak warnings after exiting product detail to go back to the product list. Here’s a screenshot:

**Comment by loremattei** [comment]
Hi there! I'm moving this to 5.1 because 5.0 has been cut. If you want this to make it to 5.0, please feel free to ping me.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductImagesFragment.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductImagesFragment.kt:90]: Yes, it's in the base class.

**Comment by 0nko** [comment]
Thank you for the review, @AmandaRiu!

> Disappearing upload thumbnail

This bug isn't the result of this PR. I can reproduce the same behavior in `develop`. I've created an issue for it so it can be fixed separately: #2828.

> Also, after completing the steps above I’m seeing LeakCanary leak warnings after exiting product detail to go back to the product list. 

I'm also seeing a leak but with a different stack trace. However, I can see the same leak in `develop`, so it's likely not caused by this PR either. I'm planning to address all memory leaks in the upcoming Hack week.

**Review by AmandaRiu — APPROVED** [review]
@0nko Since the issues I found during the PR review already exist in develop I'm going to go ahead and approve/merge this one. :shipit:

**Pull request merged**